src/extension.ts

```diff
@@ -188,18 +217,18 @@ export function hideStatusLine(): void {
 }
 
 // start the integration file
-async function runTheFile(context: vscode.Uri) {
-	await startIntegration(context)
+async function runTheFile(...args: any[]) {
+	await startIntegration(args)
 		.then( async () => await camelKIntegrationsProvider.refresh())
 		.catch ( (error) => console.log(error) );
 }
 
 // start an integration from a file
-function startIntegration(context: vscode.Uri): Promise<any> {
+function startIntegration(...args: any[]): Promise<any> {
 	return new Promise <any> ( async (resolve, reject) => {
 		setStatusLineMessageAndShow(`Starting new Apache Camel K Integration...`);
 		utils.shareMessage(mainOutputChannel, "Starting new integration via Kamel executable.");
-		await integrationutils.startIntegration(context)
+		await integrationutils.startIntegration(args)
```

Comment:
better not mix await with then/catch

Reply:
Again, bad things happen when I try to move this to try/catch. I think that the first case worked because it wasn't set up with a resolve/reject. This particular tangled combination in extension.ts is tied to that resolve/reject pattern.